packages/firebase_ai/firebase_ai/lib/src/firebase_ai.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:firebase_app_check/firebase_app_check.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_core_platform_interface/firebase_core_platform_interface.dart'
+    show FirebasePluginPlatform;
+import 'package:meta/meta.dart';
+
+import '../firebase_ai.dart';
+import 'base_model.dart';
+
+const _defaultLocation = 'us-central1';
+
+/// The entrypoint for generative models.
+class FirebaseAI extends FirebasePluginPlatform {
+  FirebaseAI._(
+      {required this.app,
+      required this.location,
+      required bool useVertexBackend,
+      this.appCheck,
+      this.auth})
+      : _useVertexBackend = useVertexBackend,
+        super(app.name, 'plugins.flutter.io/firebase_vertexai');
+
+  /// The [FirebaseApp] for this current [FirebaseAI] instance.
+  FirebaseApp app;
+
+  /// The optional [FirebaseAppCheck] for this current [FirebaseAI] instance.
+  /// https://firebase.google.com/docs/app-check
+  FirebaseAppCheck? appCheck;
+
+  /// The optional [FirebaseAuth] for this current [FirebaseAI] instance.
+  FirebaseAuth? auth;
+
+  /// The service location for this [FirebaseAI] instance.
+  String location;
+
+  final bool _useVertexBackend;
+
+  static final Map<String, FirebaseAI> _cachedInstances = {};
+
+  /// Returns an instance using a specified [FirebaseApp].
+  ///
+  /// If [app] is not provided, the default Firebase app will be used.
+  /// If pass in [appCheck], request session will get protected from abusing.
+  static FirebaseAI vertexAI({
+    FirebaseApp? app,
+    FirebaseAppCheck? appCheck,
+    FirebaseAuth? auth,
+    String? location,
+  }) {
+    app ??= Firebase.app();
+
```

Comment:
would need to apply the fix of var instanceKey = '${app.name}::googleai';

See here: https://github.com/firebase/flutterfire/pull/17319/files#diff-edacf0f74d51415959b1f96ed817e5a7cc222da1b27d3fd5904776f21a62815b

Reply:
Done.